js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -1074,6 +1075,8 @@ var EditingController = modules.ViewController.inherit((function() {
             if(deferreds.length) {
                 that._saving = true;
 
+                dataSource && dataSource._changeLoadingCount(1);
```

Comment:
Adding this logic to DataGrid will be in ten times harder. Also by your logic "hack" will be moved from DataSource to DataGrid because we use the internal loadingChanged event for showing/hiding loadPanel.

My opinion that most simple solution is most correct. I don't see problems if loadingChanged will work for inserting/updating/removing. Our widgets (DataGrid/PivotGrid/List/TileView/TreeView) use loadingChanged event for showing load indicator. If this event is designed for showing load indicator then it should be raised during long updating operations.